packages/cip2/src/selectionConstraints.ts

```diff
@@ -4,16 +4,28 @@ import { ComputeMinimumCoinQuantity, EstimateTxFee, SelectionSkeleton } from './
 import { maxBigNum } from './util';
 
 export type BuildTx = (selection: SelectionSkeleton) => Promise<CSL.Transaction>;
+
+// Review: we might want to use this for ProtocolParametersRequiredByWallet instead
```

Comment:
Something like ?

``` TypeScript
export type ProtocolParametersRequiredByWallet = ProtocolParametersRequiredBySelectionConstraints && NonNullableFields<
  Pick<
    ProtocolParametersAlonzo,
    | 'stakeKeyDeposit'
    | 'minPoolCost'
    | 'poolDeposit'
    | 'protocolVersion'
  >
>;
```

Reply:
I changed this to use `ProtocolParametersRequiredByWallet` and throwing an exception in `defaultSelectionConstraints` if any param is null

Reply:
As we want this package to be standalone I think we should still make a smaller type for coin selection though.